src/utils/database_util.py

```diff
@@ -184,3 +186,64 @@ def table_weeklychallenge_getlatest(self):
             return None
         sorted_results = sorted(results, key=lambda challenge: challenge["date"])
         return sorted_results[-1]
+
+    def table_weeklychallenge_delete(self, challenge_id: int) -> bool:
+        """
+        Deletes an item in the Weekly_Challenge database table and 
+        calls weeklyquestion_delete_by_challenge_id() to delete associated questions from
+        Weekly_Question database table
+        """
+        self.table_weeklyquestion_delete_by_challenge_id(challenge_id)
+        table = self.db.table(self.TABLE_WEEKLY_CHALLENGE)
+        results = table.remove(where("id") == challenge_id)
+        return len(results) > 0
+
+    @validate_insert(required_fields=TABLE_WEEKLY_QUESTION_FIELDS)
+    def table_weeklyquestion_insert(self, item: dict) -> bool:
+        """
+        Inserts an item into the Weekly_Question database table
+        """
+        table = self.db.table(self.TABLE_WEEKLY_QUESTION)
+
+        # Prevent duplicate ID's
+        if len(table.search(where("title_slug") == item["title_slug"])) == 0:
+            table.insert(item)
+            return True
+        return False
+
+    def table_weeklyquestion_load_by_challenge_id(self, challenge_id) -> list:
+        """
+        Loads multiple items by challenge id in the Weekly_Question database table
+        """
+        table = self.db.table(self.TABLE_WEEKLY_QUESTION)
+        questions = table.search(where("challenge_id") == challenge_id)
+        if len(questions) == 0:
+            return []
+        return questions
```

Comment:
Now that I think about it .search() already returns a list. The list will just be empty if there are no matches. So this could be greatly simplified with
```
return table.search(where("challenge_id") == challenge_id)
```

Reply:
lol, good catch!